frontend/src/views/new-project/NewProjectView.tsx

```diff
@@ -60,66 +60,101 @@ const createProjectMutation = gql`
  */
 export const NewProjectView = () => {
   const [createProject, { loading, data, error }] = useMutation(createProjectMutation);
-  const [state, dispatch] = useReducer(reducer, initialState);
-
-  const { viewState, name, nameMessage, nameMessageSeverity, message, newProjectId } = state;
+  const [{ value, context }, dispatch] = useMachine<NewProjectViewContext, NewProjectEvent>(newProjectViewMachine);
+  const { newProjectView, toast } = value as SchemaValue;
+  const { name, newProjectId, message } = context;
 
   const onNameChange = (event) => {
-    const name = event.target.value;
-    dispatch({ type: HANDLE_CHANGE_NAME__ACTION, name });
+    const value = event.target.value;
+    const changeNameEvent: ChangeNameEvent = { type: 'CHANGE_NAME', name: value };
+    dispatch(changeNameEvent);
   };
 
   const onCreateNewProject = async (event) => {
     event.preventDefault();
-
     const variables = {
       input: {
         name: name.trim(),
         visibility: 'PUBLIC',
       },
     };
+    const submitEvent: SubmitEvent = { type: 'SUBMIT' };
+    dispatch(submitEvent);
     createProject({ variables });
   };
 
   useEffect(() => {
     if (!loading && !error && data) {
       if (error) {
-        dispatch({ type: HANDLE_SUBMIT_FAILURE__ACTION, reponse: error });
+        let errorMessage = error.message;
+        const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message: errorMessage };
+        dispatch(showToastEvent);
       } else if (data) {
```

Comment:
Don't use an else if, you can have both errors and data! Ok in such a simple example there 99.999% chance that it can't occur but don't get used to this pattern. You can have both at the same time.